### PR TITLE
frontend/enh/coin-popup

### PR DESCRIPTION
**Pull request by DwivediAman-exe**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-driver/src/Screens/DriverEarningsScreen/Transformer.purs [hunk 1]
```diff
@@ -118,4 +118,4 @@ checkPopupShowToday popupType appConfig hsState = do
     _ -> ST.NO_COIN_POPUP
 
 isPopupShownToday :: String -> Boolean
-isPopupShownToday popupInfoDate = not (isToday popupInfoDate) || STR.null popupInfoDate
\ No newline at end of file
+isPopupShownToday popupInfoDate = (not (isToday popupInfoDate) || STR.null popupInfoDate)
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Screens/DriverEarningsScreen/Transformer.purs:121]: We can remove the extra brackets

File: Frontend/ui-driver/src/Helpers/Utils.js [hunk 2]
```diff
@@ -433,6 +433,18 @@ export const isYesterday = function (dateString) {
   return false;
 }
 
+export const isDaysBack = function (dateString, numOfDays) {
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Helpers/Utils.js:436]: Can we rename this function?

File: Frontend/ui-driver/src/Flow.purs [hunk 3]
```diff
@@ -3864,7 +3864,7 @@ callGetPastDaysData appConfig hsState = do
       checkCoinIsEnabled = appConfig.feature.enableYatriCoins && cityConfig.enableYatriCoins
       vehicleVariant = hsState.data.vehicleType 
       isAutoRicksaw = RC.getCategoryFromVariant vehicleVariant == Just ST.AutoCategory
-  if (isPopupShownToday coinPopupInfo.rideMoreEarnCoin && checkCoinIsEnabled && isAutoRicksaw && withinTimeRange RC.rideMoreEarnMorePopupStartTime RC.dayEndTime (convertUTCtoISC (getCurrentUTC "") "HH:mm:ss")) then do
+  if (((isPopupShownToday coinPopupInfo.rideMoreEarnCoin) && (STR.null coinPopupInfo.rideMoreEarnCoin || (runFn2 isDaysBack coinPopupInfo.rideMoreEarnCoin appConfig.coinsConfig.rideMoreEarnCoinIntervalLimit))) && checkCoinIsEnabled && isAutoRicksaw && withinTimeRange RC.rideMoreEarnMorePopupStartTime RC.dayEndTime (convertUTCtoISC (getCurrentUTC "") "HH:mm:ss")) then do
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Flow.purs:3867]: We can remove extra brackets

File: Frontend/ui-driver/src/Flow.purs [hunk 4]
```diff
@@ -3778,12 +3778,12 @@ updateBannerAndPopupFlags = do
...
         && getDriverInfoResp.mode /= Just "OFFLINE"
     
-    isCoinPopupNotShownToday = (isYesterday $ getValueToLocalStore COINS_POPUP_SHOWN_DATE) || (getValueToLocalStore COINS_POPUP_SHOWN_DATE == "__failed")
+    isCoinPopupNotShownToday = (isYesterday (getValueToLocalStore COINS_POPUP_SHOWN_DATE)) || (getValueToLocalStore COINS_POPUP_SHOWN_DATE == "__failed")
     showCoinPopup = fromMaybe 0 (fromString $ getValueToLocalStore VISITED_DRIVER_COINS_PAGE) == 0
         && appConfig.feature.enableYatriCoins
         && cityConfig.enableYatriCoins
         && toBool (getValueToLocalNativeStore IS_RIDE_ACTIVE)  == false 
-        && (fromMaybe 0 (fromString (getValueToLocalNativeStore FREE_TRIAL_DAYS)) /= 7)
+        && fromMaybe 0 (fromString $ getValueToLocalStore INTRODUCING_YATRI_POINTS_POPUP_LIMIT) < 2
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Flow.purs:3786]: Can we move this to config?

File: Frontend/ui-driver/src/Helpers/Utils.js (hunk 2 above)
Comment (DwivediAman-exe) [reply Frontend/ui-driver/src/Helpers/Utils.js]: -1

**Review by prakharritik — APPROVED** [review]

**Pull request merged**